src/tree/v2/ResourceGroupsItemCache.ts

```diff
@@ -109,8 +118,17 @@ export class ResourceGroupsItemCache {
         return currentItem;
     }
 
-    updateItemChildren(item: ResourceGroupsItem, children: ResourceGroupsItem[]): void {
+    updateItemChildren(item: ResourceGroupsItem, children: ResourceGroupsItem[]): InternalResourceGroupsItem[] {
         this.itemToChildrenCache.set(item, children);
-        children.forEach(child => this.itemToParentCache.set(child, item));
+        // cache the parent on the item
+        return children.map(child => Object.assign(child, { parent: item }));
+    }
+
+    getId(element: ResourceGroupsItem): string {
+        return '/' + this.getPathForItem(element).join('/');
+    }
+
+    isAncestorOf(element: ResourceGroupsItem, id: string): boolean {
+        return element?.isAncestorOf?.(id) || id.startsWith(this.getId(element) + '/');
```

Comment:
We should check `isAncestorOf` a little more specifically here. If it is defined but returns `false`, the `||` will still move forward with checking `id.startsWith(...)`. I'm guessing the intent is that if it is defined, then _only_ its result is considered, not the `id.startsWith()` check.

Something like this I think is more appropriate:

```typescript
if (element.isAncestorOf) { // Or: `if (typeof element.isAncestorOf === 'function')` if we need to be more resilient, but I don't think we necessarily do here
    return element.isAncestorOf(id);
} else {
    return id.startsWith(this.getId(element) + '/');
}
```

Reply:
Good catch. I think

```ts
return element.isAncestorOf?.(id) ?? id.startsWith(this.getId(element) + '/');
```

will work the same as your suggestion.

Reply:
Yup, I'd temporarily forgotten about `??`.